pbrd/pbr_nht.c

```diff
@@ -719,8 +719,51 @@ pbr_nht_individual_nexthop_gw_update(struct pbr_nexthop_cache *pnhc,
 {
 	bool is_valid = pnhc->valid;
 
-	if (!pnhi->nhr) /* It doesn't care about non-nexthop updates */
+	/*
+	 * If we have an interface down event, let's note that
+	 * it is happening and find all the nexthops that depend
+	 * on that interface.  As that if we have an interface
+	 * flapping fast enough it means that zebra might turn
+	 * those nexthop tracking events into a no-update
+	 * So let's search and do the right thing on the
+	 * interface event.
+	 */
+	if (!pnhi->nhr && pnhi->ifp) {
+		struct connected *connected;
+		struct listnode *node;
+		struct prefix p;
+
+		switch (pnhc->nexthop.type) {
+		case NEXTHOP_TYPE_BLACKHOLE:
+			goto done;
+			break;
```

Comment:
do we need the `goto` and the `break` here? I would think the `break` won't ever be called?

Reply:
I think the compiler complains if you don't have it

Reply:
I usually like to annotate cases like  this with a comment in the code like "compiler appeasement"  or something. Just to indicate that the  author is ware of it and doing it for a reason.